### PR TITLE
 adds hyperparameter models module and pep8 fixes

### PR DESCRIPTION
**Pull request by kamuda1**
@kamuda1 : All PRs should have a description. Please add one here?

### CODE REVIEW

File: annsa/learning_curve.py [hunk 1]
```diff
@@ -13,24 +13,58 @@
...
-                        # default 
-                     train_sizes = None,
-                     dense_nodes = None,
-                     weight_id = None,
-                     training_dataset_id = '../../source-interdiction/dataset_generation/testing_dataset_200keV_log10time_10000.npy',
-                     difficulty_setting = 'easy',):
-
-    
     os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
     os.environ["CUDA_VISIBLE_DEVICES"] = GPU_device_id
```
Comment (samgdotson) [inline annsa/learning_curve.py]: Shouldn't this code be under:
`if __name__== "__main__":`

Not sure what the best practice is for this, but I like separating the function definitions and executable code.

File: annsa/hyperparameter_models.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+        activation_function=choice([tf.nn.tanh, tf.nn.relu]),
+        output_function=None,
+        scaler=choice(scaler_choices))
+
+    model = DNN(model_features)
+
+    return model, model_features
+
+
+def generate_random_cnn1d_architecture(cnn_filters_choices,
```
Comment (samgdotson) [inline annsa/hyperparameter_models.py]: Isn't this function defined in `model_classes.py`?

File: annsa/hyperparameter_models.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            cnn_strides=cnn_strides,
+            pool_size=pool_size,
+            pool_strides=pool_strides,
+            dense_nodes=dense_nodes
+            )
+
+    return model_features
+
+
+def make_conv1d_model(all_keys_binarized):
```
Comment (samgdotson) [inline annsa/hyperparameter_models.py]: You already have all of these `make_model` functions defined in the tests. Would you rather have them defined here and then used by the tests?

**Review by samgdotson — CHANGES_REQUESTED** [review]
Do any of these files replace older files that are no longer used? That would be a good thing to include in this PR or your next PR.

File: annsa/learning_curve.py (hunk 1 above)
Comment (katyhuff) [reply annsa/learning_curve.py]: I agree, basically... nothing should be in global scope. If this is a packge, ALL of the global scope stuff could be induced to run as soon as one imports annsa. (yikes.)

**Review by katyhuff — CHANGES_REQUESTED** [review]

**Comment by katyhuff** [comment]
1) @kamuda1  Please don't merge your own PRs.
2) Particularly if they're failing the tests... 
3) And most of all if there are blocking review comments.

**Pull request merged**